eng/scripts/automation/GenerateAndBuildLib.ps1

```diff
@@ -324,25 +322,112 @@ function Invoke-Pack() {
     Pop-Location
 }
 function Get-ResourceProviderFromReadme($readmeFile) {
-    $readmeFileRegex = "(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    $readmeFileRegexWithSpec = "specification/(?<specName>.*)/(?<serviceType>.*)/readme.md"
-    try
-    {
-        if ($readmeFile -match $readmeFileRegexWithSpec)
-        {
-            return $matches["specName"], $matches["serviceType"]
+    $readmeFile = $readmeFile -replace "\\", "/"
+    $pathArray = $readmeFile.Split("/");
+
+    if ( $pathArray.Count -lt 3) {
+        Write-Error "Error: invalid readme file path."
+        exit 1
+    }
+
+    $specName = $pathArray[-3]
+    $serviceType = $pathArray[-2]
+    Write-Host "specName:$specName, serviceType: $serviceType"
+
+    return $specName, $serviceType
+}
+
+function Invoke-GenerateAndBuildSDK () {
+    param(
+        [string]$readmeAbsolutePath,
+        [string]$sdkRootPath,
+        [object]$generatedSDKPackages
+    )
+    $readmeFile = $readmeAbsolutePath -replace "\\", "/"
+    Write-Host "readmeFile:$readmeFile"
+    $service, $serviceType = Get-ResourceProviderFromReadme $readmeFile
+    Write-Host "service:$service, serviceType:$serviceType"
+    
+    if (!(Test-Path -Path $readmeFile)) {
+        Write-Error "readme file does not exist."
+        exit 1
+    }
+    
+    $packagesToGen = @()
+    $newpackageoutput = "newPackageOutput.json"
+    if ( $serviceType -eq "resource-manager" ) {
+        Write-Host "Generate resource-manager SDK client library."
+        $package = $service
```

Comment:
Why would the package and service be the same here? Is this some special case where we expect mgmt to only have one package per service? If so how to we end up finding the namespace name from the service?

Reply:
By default, the package of mgmt is the same as service name, and same as the serviceDirectory in azure-sdk-for-net, the  exception will be recorded in $packageNameHash